pkgs/development/libraries/cosmopolitan/default.nix

```diff
@@ -1,68 +1,87 @@
-{ lib, gcc9Stdenv, fetchFromGitHub, runCommand, cosmopolitan }:
+{ lib, stdenv, fetchFromGitHub, runCommand, unzip, cosmopolitan,bintools-unwrapped }:
 
-gcc9Stdenv.mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "cosmopolitan";
-  version = "0.3";
+  version = "unstable-2022-03-22";
 
   src = fetchFromGitHub {
     owner = "jart";
     repo = "cosmopolitan";
-    rev = version;
-    sha256 = "sha256-OVdOObO82W6JN63OWKHaERS7y0uvgxt+WLp6Y0LsmJk=";
+    rev = "5022f9e9207ff2b79ddd6de6d792d3280e12fb3a";
+    sha256 = "sha256-UjL4wR5HhuXiQXg6Orcx2fKiVGRPMJk15P779BP1fRA=";
   };
 
+  patches = [
+    ./ioctl.patch        # required /dev/tty
+  ];
+  /* WIP Changes needed to work with Nixpkgs native gcc
+  OVERRIDE_CFLAGS = "-Wno-old-style-definition";
+   preBuild2 = ''
+    makeFlagsArray=(
+      SHELL=/bin/sh
+      AS=${stdenv.cc.targetPrefix}as
+      CC=${stdenv.cc.targetPrefix}gcc
+      GCC=${stdenv.cc.targetPrefix}gcc
+      CXX=${stdenv.cc.targetPrefix}g++
+      LD=${stdenv.cc.targetPrefix}ld
+      OBJCOPY=${stdenv.cc.targetPrefix}objcopy
+      ADDR2LINE=${bintools-unwrapped}/bin/addr2line
+      "MKDIR=mkdir -p"
+      )
+   '';
+  rm -r third_party/gcc
+  */
```

Comment:
`preBuild2` isn't a thing and the `rm` line is ambiguous since it's not marked as `prePatch`. I agree with the intent but I think the commit history is enough to preserve this information as of now.